scripts/template_upload_scvmm.py

```diff
@@ -138,95 +138,128 @@ def make_kwargs_scvmm(cfme_data, provider, image_url, template_name):
     return final_kwargs
 
 
+def create_template(provider, **kwargs):
+    mgmt_sys = cfme_data['management_systems'][provider]
+    host_fqdn = mgmt_sys['hostname_fqdn']
+    creds = credentials[mgmt_sys['credentials']]
+
+    kwargs = make_kwargs_scvmm(
+        cfme_data, provider, kwargs.get('image_url'), kwargs.get('template_name'))
+    check_kwargs(**kwargs)
+
+    # For powershell to work, we need to extract the User Name from the Domain
+    user = creds['username'].split('\\')
+    if len(user) == 2:
+        username_powershell = user[1]
+    else:
+        username_powershell = user[0]
+
+    username_scvmm = creds['domain'] + "\\" + creds['username']
+
+    scvmm_args = {
+        "hostname": mgmt_sys['ipaddress'],
+        "username": username_powershell,
+        "password": creds['password'],
+        "domain": creds['domain'],
+        "provisioning": mgmt_sys['provisioning']
+    }
+    client = SCVMMSystem(**scvmm_args)
+
+    url = kwargs.get('image_url')
+
+    # Template name equals either user input of we extract the name from the url
+    new_template_name = kwargs.get('template_name')
+    if new_template_name is None:
+        new_template_name = os.path.basename(url)[:-4]
+
+    logger.info("SCVMM:%s Make Template out of the VHD %s", provider, new_template_name)
+
+    # use_library is either user input or we use the cfme_data value
+    library = kwargs.get('library', mgmt_sys['template_upload'].get('vhds', None))
+
+    logger.info("SCVMM:%s Template Library: %s", provider, library)
+
+    #  The VHD name changed, match the template_name.
+    new_vhd_name = new_template_name + '.vhd'
+
+    network = mgmt_sys['template_upload'].get('network', None)
```

Comment:
**REQUIRED**
No need to define `None` as an argument in `get`. It's returned by default.